src/adapter/threads.ts

```diff
@@ -885,9 +887,32 @@ export class Thread implements IVariableStoreDelegate {
         };
       default:
         if (event.hitBreakpoints && event.hitBreakpoints.length) {
-          const isStopOnEntry =
+          let isStopOnEntry = false; // By default we assume breakpoints aren't stop on entry
+          if (
             event.hitBreakpoints.length === 1 &&
-            this._delegate.entryBreakpointId === event.hitBreakpoints[0];
+            this._delegate.entryBreakpoint?.cdtpId === event.hitBreakpoints[0]
+          ) {
+            isStopOnEntry = true; // But if it matches the entry breakpoint id, then it's probably stop on entry
+            const entryBreakpointAbsolutePath = fileUrlToAbsolutePath(
+              this._delegate.entryBreakpoint.path,
+            );
+            const entryBreakpointSource = this._sourceContainer.source({
+              path: entryBreakpointAbsolutePath,
+            });
+
+            if (entryBreakpointSource !== undefined) {
+              const entryBreakpointLocations = this._sourceContainer.currentSiblingUiLocations({
+                lineNumber: 1,
+                columnNumber: 1,
+                source: entryBreakpointSource,
+              });
+
+              // But if there is a user breakpoint on the same location that the stop on entry breakpoint, then we consider it an user breakpoint
```

Comment:
If there is a user breakpoint at the same spot, will `event.hitBreakpoints` contain all of them?

Reply:
In the cases I tested it just contains the entry breakpoint. I wrote the code as-is in case we run into a case where we find more than 1 breakpoint here.